From: Huy Nguyen <huyn@mellanox.com>
Date: Wed, 9 Aug 2017 10:03:40 -0500
Subject: net/mlx5: Remove the flag MLX5_INTERFACE_STATE_SHUTDOWN
Patch-mainline: v4.13
Git-commit: 10a8d00707082955b177164d4b4e758ffcbd4017
References: bsc#966170 FATE#320225 bsc#966172 FATE#320226

MLX5_INTERFACE_STATE_SHUTDOWN is not used in the code.

Fixes: 5fc7197d3a25 ("net/mlx5: Add pci shutdown callback")
Signed-off-by: Huy Nguyen <huyn@mellanox.com>
Reviewed-by: Daniel Jurgens <danielj@mellanox.com>
Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/mellanox/mlx5/core/main.c |    2 --
 include/linux/mlx5/driver.h                    |    1 -
 2 files changed, 3 deletions(-)

--- a/drivers/net/ethernet/mellanox/mlx5/core/main.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/main.c
@@ -1512,8 +1512,6 @@ static void shutdown(struct pci_dev *pde
 	struct mlx5_priv *priv = &dev->priv;
 
 	dev_info(&pdev->dev, "Shutdown was called\n");
-	/* Notify mlx5 clients that the kernel is being shut down */
-	set_bit(MLX5_INTERFACE_STATE_SHUTDOWN, &dev->intf_state);
 	mlx5_unload_one(dev, priv, false);
 	mlx5_pci_disable_device(dev);
 }
--- a/include/linux/mlx5/driver.h
+++ b/include/linux/mlx5/driver.h
@@ -666,7 +666,6 @@ enum mlx5_device_state {
 
 enum mlx5_interface_state {
 	MLX5_INTERFACE_STATE_UP = BIT(0),
-	MLX5_INTERFACE_STATE_SHUTDOWN = BIT(1),
 };
 
 enum mlx5_pci_status {
